src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs

```diff
@@ -72,25 +72,9 @@ public async Task<ClientCredentialsToken> RequestTokenAsync(string clientName, C
 
     private static void ThrowOnIdentityModelError(string clientName, TokenResponse response)
     {
-        // Checking this error type first helps us preserve the original exception
-        if (response is { ErrorType: ResponseErrorType.Exception, Exception: { } exception })
-        {
-            throw new ClientCredentialsException(GetErrorMessage(clientName, response), exception);
-        }
-
-        // TokenResponse.IsError and TokenResponse.Error implementations are incomplete but we handle the missing use cases below
-        // https://github.com/IdentityModel/IdentityModel/blob/6.0.0/src/Client/Messages/ProtocolResponse.cs#L190
         if (response.IsError)
         {
-            throw new ClientCredentialsException(GetErrorMessage(clientName, response));
-        }
-
-        switch (response.ErrorType)
```

Comment:
response.IsError is better now and this code path couldn't be reached anymore
Before: https://github.com/IdentityModel/IdentityModel/blob/6.0.0/src/Client/Messages/ProtocolResponse.cs#L190
Now: https://github.com/IdentityModel/IdentityModel/blob/9ce1195f6cb07a3ed80c66ef8d87489e08aac987/src/Client/Messages/ProtocolResponse.cs#L175